pkg/operator/rbaccontroller/rbaccontroller.go

```diff
@@ -0,0 +1,78 @@
+package rbaccontroller
+
+import (
+	"context"
+
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apiserver/pkg/authentication/user"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/klog"
+)
+
+type RBACController struct {
+	ctx           context.Context
+	kubeClient    kubernetes.Interface
+	eventRecorder events.Recorder
+}
+
+func NewRBACController(kubeClient kubernetes.Interface, informerFactory informers.SharedInformerFactory, recorder events.Recorder) factory.Controller {
+
+	informer := informerFactory.
+		Rbac().
+		V1().
+		ClusterRoleBindings().
+		Informer()
+
+	c := &RBACController{
+		kubeClient:    kubeClient,
+		eventRecorder: recorder,
+	}
+
+	return factory.New().
+		WithInformers(informer).
+		WithSync(c.sync).
+		ToController("RBACController", c.eventRecorder)
+}
+
+func (c *RBACController) sync(ctx context.Context, syncContext factory.SyncContext) error {
+	clusterRoleBindings := []string{"system:discovery", "system:openshift:discovery", "cluster-status-binding"}
+
+	var errs []error
+
+	for _, clusterRoleBinding := range clusterRoleBindings {
+		if err := c.reconcileClusterRoleBinding(clusterRoleBinding); err != nil {
+			errs = append(errs, err)
+		}
+	}
+
+	return utilerrors.NewAggregate(errs)
+}
+
+func (c *RBACController) reconcileClusterRoleBinding(name string) error {
```

Comment:
This could create a new problem for us.   You need to have a way to opt-out of having the clusterrolebinding tightened.  We use an annotation in openshift/kube for "don't reconcile"

Reply:
I see what you are saying. I will add a new annotation for these clusterrolebindings